homeassistant/components/homekit_controller/humidifier.py

```diff
@@ -0,0 +1,224 @@
+"""Support for HomeKit Controller humidifier."""
+from typing import Any, Dict, List, Optional
+
+from aiohomekit.model.characteristics import CharacteristicsTypes
+
+from homeassistant.components.humidifier import HumidifierEntity
+from homeassistant.components.humidifier.const import (
+    ATTR_AVAILABLE_MODES,
+    ATTR_HUMIDITY,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MIN_HUMIDITY,
+    ATTR_MODE,
+    DEVICE_CLASS_DEHUMIDIFIER,
+    DEVICE_CLASS_HUMIDIFIER,
+    MODE_AUTO,
+    SUPPORT_MODES,
+)
+from homeassistant.core import callback
+
+from . import KNOWN_DEVICES, HomeKitEntity
+
+SUPPORT_FLAGS = 0
+
+ATTR_DEHUMIDIFIER_THRESHOLD = "dehumidifier_threshold"
+ATTR_HUMIDIFIER_THRESHOLD = "humidifier_threshold"
+ATTR_CURRENT_HUMIDITY = "current_humidity"
+
+MODE_OFF = "off"
+MODE_HUMIDIFYING = "humidifying"
+MODE_DEHUMIDIFYING = "dehumidifying"
+
+HK_MODE_TO_HA = {
+    0: MODE_OFF,
```

Comment:
For newer additions i've been trying to define enums on the aiohomekit side so that non-HomeKit aware Home Assistant reviewers can see a bit better that something makes sense. They are defined [here](https://github.com/Jc2k/aiohomekit/blob/main/aiohomekit/model/characteristics/const.py).

Reply:
Thanks 😊